sickbeard/common.py

```diff
@@ -512,39 +512,80 @@ def statusFromName(name, assume=True, anime=False):
                         ANY: "Any"}
 
 
-# pylint: disable=R0903,C1001
-class StatusStrings:
+class StatusStrings(dict):
+    """
+    Dictionary containing strings for status codes
+
+    Keys must be convertible to int or a ValueError will be raised.  This is intentional to match old functionality until
+    the old StatusStrings is fully deprecated, then we will raise a KeyError instead, where appropriate.
+
+    Membership checks using __contains__ (i.e. 'x in y') do not raise a ValueError to match expected dict functionality
+    """
+    # todo: Deprecate StatusStrings().statusStrings and use StatusStrings() directly
+    # todo: Deprecate .has_key and switch to 'x in y'
+    # todo: Switch from raising ValueError to a saner KeyError
+    # todo: Raise KeyError when unable to resolve a missing key instead of returning ''
+    # todo: Make key of None match dict() functionality
     def __init__(self):
-        self.statusStrings = {UNKNOWN: "Unknown",
-                              UNAIRED: "Unaired",
-                              SNATCHED: "Snatched",
-                              DOWNLOADED: "Downloaded",
-                              SKIPPED: "Skipped",
-                              SNATCHED_PROPER: "Snatched (Proper)",
-                              WANTED: "Wanted",
-                              ARCHIVED: "Archived",
-                              IGNORED: "Ignored",
-                              SUBTITLED: "Subtitled",
-                              FAILED: "Failed",
-                              SNATCHED_BEST: "Snatched (Best)"}
-
-    def __getitem__(self, key):
-        key = int(key)
-        if key in Quality.DOWNLOADED + Quality.SNATCHED + Quality.SNATCHED_PROPER + Quality.SNATCHED_BEST + Quality.ARCHIVED:
+        """
+        Initialize status strings
+        """
+        self[UNKNOWN] = "Unknown"
+        self[UNAIRED] = "Unaired"
+        self[SNATCHED] = "Snatched"
+        self[DOWNLOADED] = "Downloaded"
+        self[SKIPPED] = "Skipped"
+        self[SNATCHED_PROPER] = "Snatched (Proper)"
+        self[WANTED] = "Wanted"
+        self[ARCHIVED] = "Archived"
+        self[IGNORED] = "Ignored"
+        self[SUBTITLED] = "Subtitled"
+        self[FAILED] = "Failed"
+        self[SNATCHED_BEST] = "Snatched (Best)"
+        self.statusStrings = dict(self)  # for backwards compatibility
+
+    def __missing__(self, key):
+        """
+        If the key is not found, search for the missing key in qualities
+
+        Keys must be convertible to int or a ValueError will be raised.  This is intentional to match old functionality until
+        the old StatusStrings is fully deprecated, then we will raise a KeyError instead, where appropriate.
+        """
+        key = int(key)  # This will raise a ValueError if the key can't be converted to int
+
+        if key in self.keys() + Quality.DOWNLOADED + Quality.SNATCHED + Quality.SNATCHED_PROPER + Quality.SNATCHED_BEST + Quality.ARCHIVED:
```

Comment:
we already know it isnt in self.keys() here, or __missing__ wouldnt have been called